### PR TITLE
Use object informers instead of listers

### PR DESCRIPTION
**Pull request by cpuguy83**
We'll need these informers for
https://github.com/virtual-kubelet/virtual-kubelet/pull/586

Changing this now means we don't need to make API changes later.

### CODE REVIEW

File: node/podcontroller.go [hunk 1]
```diff
@@ -126,17 +129,17 @@ func NewPodController(cfg PodControllerConfig) (*PodController, error) {
 	if cfg.PodInformer == nil {
 		return nil, errdefs.InvalidInput("must set informer")
 	}
-	if cfg.ConfigMapLister == nil {
+	if cfg.ConfigMapInformer == nil {
 		return nil, errdefs.InvalidInput("must set config map lister")
```
Comment (pires) [inline node/podcontroller.go]: Error definition is wrong.

Comment (pires) [reply node/podcontroller.go]: Same for all previous lister validations below.

Comment (pires) [reply node/podcontroller.go]: I don't even know if it makes sense to log it like this.

Comment (cpuguy83) [reply node/podcontroller.go]: Updated

**Pull request merged**